pkg/webhooks/pytorch/pytorchjob_webhook.go

```diff
@@ -93,13 +93,22 @@ func validatePyTorchJob(oldJob, newJob *trainingoperator.PyTorchJob) (admission.
 func validateSpec(spec trainingoperator.PyTorchJobSpec) (admission.Warnings, field.ErrorList) {
 	var allErrs field.ErrorList
 	var warnings admission.Warnings
-
-	if spec.ElasticPolicy != nil && spec.ElasticPolicy.NProcPerNode != nil {
-		elasticNProcPerNodePath := specPath.Child("elasticPolicy").Child("nProcPerNode")
-		nprocPerNodePath := specPath.Child("nprocPerNode")
-		warnings = append(warnings, fmt.Sprintf("%s is deprecated, use %s instead", elasticNProcPerNodePath.String(), nprocPerNodePath.String()))
-		if spec.NprocPerNode != nil {
-			allErrs = append(allErrs, field.Forbidden(elasticNProcPerNodePath, fmt.Sprintf("must not be used with %s", nprocPerNodePath)))
+	if spec.ElasticPolicy != nil {
+		workerReplicaSpec, ok := spec.PyTorchReplicaSpecs[trainingoperator.PyTorchJobReplicaTypeWorker]
+		workerPath := specPath.Child("pytorchReplicaSpecs").Child("Worker")
+		if !ok {
+			allErrs = append(allErrs, field.Required(workerPath, "must be configured if elastic policy is used"))
+		} else if workerReplicaSpec.Replicas != nil && int(*workerReplicaSpec.Replicas) < 1 {
```

Comment:
We added this validation to cover the case where replicas is explicitly set to 0 since the default is only set when it is nil https://github.com/kubeflow/training-operator/blob/master/pkg/apis/kubeflow.org/v1/defaulting_utils.go#L43

Reply:
Maybe we should have a separate validation that disallow users to set Replicas < 1 since it is invalid option ?

Reply:
Yes, that makes sense- we will update it to a separate validation that won't depend on whether elastic policy is set. Is it okay to have both validations in the same PR or would you prefer we separate that one into its own PR?

Reply:
I think, it is fine to add this simple check in this PR if you can!

Reply:
@andreyvelich, we added the separate validation for worker replicas being at least 1. I also realized that we had `worker` as an attribute, instead of a key in the `pytorchReplicaSpecs` map, in the error field path so we fixed it to use the map representation. It should be good to re-review now